sdk/go/common/resource/plugin/langruntime.go

```diff
@@ -16,22 +16,97 @@ package plugin
 
 import (
 	"context"
+	"fmt"
 	"io"
+	"path/filepath"
 
 	"github.com/blang/semver"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	structpb "google.golang.org/protobuf/types/known/structpb"
 )
 
+// ProgramInfo contains minimal information about the program to be run.
+type ProgramInfo struct {
+	root       string
+	program    string
+	entryPoint string
+	options    map[string]any
+}
+
+func NewProgramInfo(rootDirectory, programDirectory, entryPoint string, options map[string]any) ProgramInfo {
+	isValidPath := func(path string) bool {
+		return filepath.IsLocal(path) || filepath.IsAbs(path)
+	}
+
+	if !isValidPath(rootDirectory) {
+		panic(fmt.Sprintf("rootDirectory '%s' is not a valid path when creating ProgramInfo", rootDirectory))
+	}
+
+	if !isValidPath(programDirectory) {
+		panic(fmt.Sprintf("programDirectory '%s' is not a valid path when creating ProgramInfo", programDirectory))
+	}
+
+	if !isValidPath(entryPoint) && entryPoint != "." {
```

Comment:
This should have an extra check that only names are used here "main.ts" is fine "path/to/main.ts" is not.

Reply:
Done ✅